### PR TITLE
Store contig names in a `contig_id` array

### PR DESCRIPTION
**Pull request by tomwhite**
Proposed changes to support https://github.com/pystatgen/sgkit/issues/1035

### CODE REVIEW

File: vcf_zarr_spec.md [hunk 1]
```diff
@@ -117,6 +114,24 @@ A **Genotype (GT) field** is stored as a three-dimensional Zarr array at a path
...
 
+### Contig information
+
+Contig IDs are stored in a one-dimensional Zarr array at a path with name `contig_id`, of shape `(contigs)`, dimension names `[contigs]`, and with dtype `str`. The `contig_id` array plays the same role as the dictionary of contigs in BCF, providing a way of encoding a contig (in the `variant_contig` array) with a (zero-based) integer offset into the `contig_id` array.
+
+Contig lengths are optional, and if present are stored in a one-dimensional Zarr array at a path with name `contig_length`, of shape `(contigs)`, dimension names `[contigs]`, and with dtype `int`.
+
+### Filter information
+
+Sample IDs are stored in a one-dimensional Zarr array at a path with name `filter_id`, of shape `(filters)`, dimension names `[filters]`, and with dtype `str`. Filters must appear in the same order as specified in the header, except for `PASS`, which is always first.
```
Comment (jeromekelleher) [inline vcf_zarr_spec.md:125]: Should be "Filters are stored..."?

**Review by jeromekelleher — COMMENTED** [review]
LGTM, one typo

**Review by benjeffery — APPROVED** [review]
LGTM modulo JK's comment.

**Comment by benjeffery** [comment]
One more thought - I see #1054 introduces a `source` attribute -  should that be codified here?

**Comment by tomwhite** [comment]
> One more thought - I see #1054 introduces a `source` attribute - should that be codified here?

The `source` attribute was being used in another code path, the change in https://github.com/pystatgen/sgkit/pull/1054 fixes where it was missing. But I agree we should document the convention (opened #6).

**Pull request merged**